modules/services/syncthing.nix

```
@@ -211,14 +211,25 @@ let
     ''
   );
 
-  defaultSyncthingArgs = [
-    "${syncthing}"
-    "-no-browser"
-    "-no-restart"
-    "-no-upgrade"
-    "-gui-address=${if isUnixGui then "unix://" else ""}${cfg.guiAddress}"
-    "-logflags=0"
-  ];
+  defaultSyncthingArgs =
+    if lib.versionAtLeast (lib.versions.major cfg.package.version) "2" then
+      [
+        "${syncthing}"
+        "serve"
+        "--no-browser"
+        "--no-restart"
+        "--no-upgrade"
+        "--gui-address=${if isUnixGui then "unix://" else ""}${cfg.guiAddress}"
+      ]
+    else
+      [
+        "${syncthing}"
+        "-no-browser"
+        "-no-restart"
+        "-no-upgrade"
+        "-gui-address=${if isUnixGui then "unix://" else ""}${cfg.guiAddress}"
+        "-logflags=0"
+      ];
```

```suggestion
  defaultSyncthingArgs = [
    "${syncthing}"
  ]
  ++ lib.optionals (lib.versionAtLeast (lib.versions.major cfg.package.version) "2") [
    "serve"
  ]
  ++ [
    "--no-browser"
    "--no-restart"
    "--no-upgrade"
    "--gui-address=${if isUnixGui then "unix://" else ""}${cfg.guiAddress}"
  ]
  ++ lib.optionals (lib.versionOlder (lib.versions.major cfg.package.version) "2") [
    "-logflags=0"
  ];
```
Could simplify with optionals ?